lib/exgboost/parameters.ex

```diff
@@ -0,0 +1,939 @@
+defmodule Exgboost.Parameter do
+  alias __MODULE__
+  @enforce_keys [:validation]
+  defstruct [:validation, :default, :alter, reqs: []]
+
+  def parameters do
```

Comment:
I would not do this. I think it's easier instead to just use pattern matching on a function clause. As an example, you can do something like:

```elixir
defp validate_option!(:verbosity, value) do
  ...
end
```

And implement the validation for each parameter. Another alternative is to use nimble options: https://hexdocs.pm/nimble_options/NimbleOptions.html

I would recommend nimble options

Reply:
Yes, I think nimble options is exactly what I'm looking for and only after I saw this I realized I've used them before with Scholar